tests/conftest.py

```diff
@@ -24,68 +23,37 @@
 import app
 from core.authentication import AuthenticationHelper
 
-
-class Caption:
-    def __init__(self, text, highlights=None, additional_properties=None):
-        self.text = text
-        self.highlights = highlights or []
-        self.additional_properties = additional_properties or {}
-
-
-class AsyncPageIterator:
-    def __init__(self, data):
-        self.data = data
-
-    def __aiter__(self):
-        return self
-
-    async def __anext__(self):
-        if not self.data:
-            raise StopAsyncIteration
-        return self.data.pop(0)  # This should be a list of dictionaries.
-
-
-class AsyncSearchResultsIterator:
-    def __init__(self):
-        self.data = [
-            [
-                {
-                    "sourcepage": "Benefit_Options-2.pdf",
-                    "sourcefile": "Benefit_Options.pdf",
-                    "content": "There is a whistleblower policy.",
-                    "embeddings": [],
-                    "category": None,
-                    "id": "file-Benefit_Options_pdf-42656E656669745F4F7074696F6E732E706466-page-2",
-                    "@search.score": 0.03279569745063782,
-                    "@search.reranker_score": 3.4577205181121826,
-                    "@search.highlights": None,
-                    "@search.captions": [Caption("Caption: A whistleblower policy.")],
-                },
-            ]
-        ]
-
-    def __aiter__(self):
-        return self
-
-    async def __anext__(self):
-        if not self.data:
-            raise StopAsyncIteration
-        return AsyncPageIterator(self.data.pop(0))
-
-    def by_page(self):
-        return self
+from .mocks import (
+    MockAsyncSearchResultsIterator,
+    MockAzureCredential,
+    MockBlobClient,
+    MockKeyVaultSecretClient,
+    MockResponse,
+    mock_computervision_response,
+)
 
 
 async def mock_search(self, *args, **kwargs):
     self.filter = kwargs.get("filter")
-    return AsyncSearchResultsIterator()
+    return MockAsyncSearchResultsIterator(kwargs.get("search_text"), kwargs.get("vectors"))
 
 
 @pytest.fixture
 def mock_get_secret(monkeypatch):
     monkeypatch.setattr(SecretClient, "get_secret", MockKeyVaultSecretClient().get_secret)
 
 
+@pytest.fixture
+def mock_compute_embeddings_call(monkeypatch):
+    def mock_post(*args, **kwargs):
+        if kwargs.get("url").endswith("computervision/retrieval:vectorizeText"):
+            return mock_computervision_response()
+        else:
+            raise Exception("Unexpected URL for mock call to ClientSession.post()")
+
+    monkeypatch.setattr(aiohttp.ClientSession, "post", mock_post)
```

Comment:
We need to be careful of mocking ClientSession.post() as we use it in two places, for computervision and for ADLS. Fortunately their tests don't use same mocks, but could bite us in future.